### PR TITLE
Conditions on podupdater function of kubernetes autodiscovery

### PR DESCRIPTION
**Pull request by gizas**
- Enhancement

**[Needs https://github.com/elastic/elastic-agent-autodiscover/pull/71 first to be merged and update go.mod file]**

## Proposed commit message

- WHAT: Update the PodUpdater fucntion with additonal checks before actaully triggering Pod watcher restarts
- WHY:  There where node and namespace events that dont always inlcude Metadata changes. We want only update events that change labels and annotations of node and namespace to trigger pod updates. This leads to bettwr management of watchers

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

See info https://github.com/elastic/beats/issues/37338

## Related issues

- Relates https://github.com/elastic/beats/issues/37338
- Relates https://github.com/elastic/elastic-agent-autodiscover/pull/71

### CODE REVIEW

**Comment by gizas** [comment]
@elastic/elastic-agent-data-plane team can I have a review here please as codeowners?

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -151,6 +151,8 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - elasticsearch output now supports `idle_connection_timeout`. {issue}35616[35615] {pull}36843[36843]
 - Upgrade golang/x/net to v0.17.0. Updates the publicsuffix table used by the registered_domain processor. {pull}36969[36969]
 Setting environmental variable ELASTIC_NETINFO:false in Elastic Agent pod will disable the netinfo.enabled option of add_host_metadata processor
+- Upgrade to Go 1.20.11. {pull}37123[37123]
```
Comment (michalpristas) [inline CHANGELOG.next.asciidoc:154]: not related to this change

**Review by michalpristas — APPROVED** [review]

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -137,9 +137,12 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
...
```
Comment (michalpristas) [inline CHANGELOG.next.asciidoc:140]:
```suggestion
```

File: CHANGELOG.next.asciidoc [hunk 3]
```diff
@@ -137,9 +137,12 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
...
+- Upgrade to Go 1.21.5. {pull}37550[37550]
```
Comment (michalpristas) [inline CHANGELOG.next.asciidoc:141]:
```suggestion
```

**Review by pierrehilbert — APPROVED** [review]

**Review by belimawr — APPROVED** [review]

**Review by MichaelKatsoulis — APPROVED** [review]

**Pull request merged**